Description: Restore access to cupsd.conf and logfiles, fix regression
 introduced by STR: #4455
Author: Tim Waugh <twaugh@redhat.com>
Bug-Upstream: https://www.cups.org/str.php?L4461
Bug-Debian: https://bugs.debian.org/757964
Last-Update: 2014-08-12
--- a/scheduler/client.c
+++ b/scheduler/client.c
@@ -3360,8 +3360,18 @@
 
   if (!status && !(filestats->st_mode & S_IROTH))
   {
-    cupsdLogMessage(CUPSD_LOG_INFO, "[Client %d] Files/directories such as \"%s\" must be world-readable.", con->http.fd, filename);
-    return (NULL);
+   /*
+    * The exception is for cupsd.conf and log files for
+    * authenticated access.
+    */
+
+    if ((strncmp(con->uri, "/admin/conf/cupsd.conf", 22) &&
+        strncmp(con->uri, "/admin/log/", 11)) ||
+       cupsdIsAuthorized(con, NULL) != HTTP_OK)
+    {
+      cupsdLogMessage(CUPSD_LOG_INFO, "[Client %d] Files/directories such as \"%s\" must be world-readable.", con->http.fd, filename);
+      return (NULL);
+    }
   }
 
  /*
